gui/src/installer/step/bitcoind.rs

```diff
@@ -658,6 +658,13 @@ impl Step for InternalBitcoindStep {
                         bitcoind.stop();
                     }
                     self.internal_bitcoind = None;
+                    if let Some(download) = self.exe_download.as_ref() {
+                        // Clear exe_download if not Finished.
+                        if let DownloadState::Finished { .. } = download.state {
```

Comment:
Note for the future: i think these types of conditions are more clearly expressed using the `matches!` macro